gen/field_test.go

```diff
@@ -76,3 +78,54 @@ func TestFieldLabelConflict(t *testing.T) {
 		})
 	}
 }
+
+func TestCompileJSONTag(t *testing.T) {
+	tests := []struct {
+		desc             string
+		required         bool
+		options          []string
+		isOmitemptyAdded bool
+	}{
+		{
+			desc:             "for optional fields",
```

Comment:
nit: "for" prefix unnecessary for test case descriptions.

```suggestion
			desc:             "optional fields",
```